fast64_internal/sm64/sm64_level_writer.py

```diff
@@ -697,279 +737,272 @@ def __init__(self):
-        splineFile = open(os.path.join(areaDir, "spline.inc.c"), "w", newline="\n")
-        splinesC = area.to_c_splines()
-        splineFile.write(splinesC.source)
-        splineFile.close()
-        levelDataString += '#include "levels/' + levelName + "/" + areaName + '/spline.inc.c"\n'
-        headerString += splinesC.header
-
-    cameraVolumeString += "\tNULL_TRIGGER\n};"
 
-    # Generate levelscript string
-    compressionFmt = bpy.context.scene.compressionFormat
-    replaceSegmentLoad(prevLevelScript, f"_{levelName}_segment_7", f"LOAD_{compressionFmt.upper()}", 0x07)
-    if usesEnvFX:
-        replaceSegmentLoad(prevLevelScript, f"_effect_{compressionFmt}", f"LOAD_{compressionFmt.upper()}", 0x0B)
-    if not obj.useBackgroundColor:
-        segment = ""
-        if obj.background == "CUSTOM":
-            segment = obj.fast64.sm64.level.backgroundSegment
-        else:
-            segment = backgroundSegments[obj.background] + "_skybox"
+    level_data.camera_data += "\tNULL_TRIGGER\n};"
 
-        replaceSegmentLoad(prevLevelScript, f"_{segment}_{compressionFmt}", f"LOAD_{compressionFmt.upper()}", 0x0A)
-    # actor groups
-    if obj.fast64.sm64.segment_loads.seg5_enum != "Do Not Write":
-        replaceSegmentLoad(
-            prevLevelScript,
-            f"_{obj.fast64.sm64.segment_loads.seg5}_{compressionFmt}",
-            f"LOAD_{compressionFmt.upper()}",
-            0x05,
-        )
-        replaceSegmentLoad(prevLevelScript, f"_{obj.fast64.sm64.segment_loads.seg5}_geo", "LOAD_RAW", 0x0C)
-    if obj.fast64.sm64.segment_loads.seg6_enum != "Do Not Write":
-        replaceSegmentLoad(
-            prevLevelScript,
-            f"_{obj.fast64.sm64.segment_loads.seg6}_{compressionFmt}",
-            f"LOAD_{compressionFmt.upper()}",
-            0x06,
-        )
-        replaceSegmentLoad(prevLevelScript, f"_{obj.fast64.sm64.segment_loads.seg6}_geo", "LOAD_RAW", 0x0D)
-    replaceScriptLoads(prevLevelScript, obj)
-    levelscriptString = prevLevelScript.to_c(areaString)
+    # Generate levelscript string
+    export_level_script_c(obj, prev_level_script, level_name, level_data, level_dir, uses_env_fx)
 
     if bpy.context.scene.exportHiddenGeometry:
         restoreHiddenState(hiddenState)
 
     # Remove old areas.
-    for f in os.listdir(levelDir):
-        if re.search("area\_\d+", f):
+    for folder in os.listdir(level_dir):
+        if re.search("area\_\d+", folder):
             existingArea = False
-            for index, areaObj in areaDict.items():
-                if f == "area_" + str(index):
+            for index, areaObj in area_dict.items():
+                if folder == f"area_{index}":
                     existingArea = True
             if not existingArea:
-                shutil.rmtree(os.path.join(levelDir, f))
+                shutil.rmtree(os.path.join(level_dir, folder))
+
+    def include_proto(file_name):
+        return f'#include "levels/{level_name}/{file_name}/"\n'
```

Comment:
there should not be a / after the file name (aside from being weird it causes a compilation error so there is that)

Reply:
So is this